app/services/withdraw_participant.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+class WithdrawParticipant
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  attribute :cpd_lead_provider
+  attribute :participant_id
+  attribute :reason
+  attribute :course_identifier
+  attribute :force_training_status_change
+
+  validate :participant_has_participant_profile
+  validates :cpd_lead_provider, induction_record: true
+  validates :reason,
+            presence: { message: I18n.t(:missing_reason) },
+            inclusion: {
+              in: ->(klass) { klass.participant_profile.class::WITHDRAW_REASONS },
+              message: I18n.t(:invalid_reason),
+            }, if: ->(klass) { klass.participant_profile.present? }
+  validates :course_identifier, course: true, presence: { message: I18n.t(:missing_course_identifier) }
+  validate :not_already_withdrawn
+
+  def call
+    ActiveRecord::Base.transaction do
+      create_withdrawn_participant_profile_state!
+      update_withdrawn_induction_record!
+
+      participant_profile.training_status_withdrawn!
+    end
+
+    unless participant_profile.npq?
```

Comment:
I was wondering if you could add a test for this?

Reply:
Hi @cwrw tksss 👍 😄  it's been addressed in [51b8090](https://github.com/DFE-Digital/early-careers-framework/pull/2639/commits/51b8090cf4b44933d43fcf8e0ccc693fc7ae0838)